libraries/aws/jobs/src/aws_iot_jobs_serialize.c

```diff
@@ -492,33 +489,35 @@ static AwsIotJobsError_t _generateGetPendingRequest( const AwsIotJobsRequestInfo
     if( pJobsRequest == NULL )
     {
         IotLogError( "No memory for Jobs GET PENDING request." );
-
-        IOT_SET_AND_GOTO_CLEANUP( AWS_IOT_JOBS_NO_MEMORY );
+        status = AWS_IOT_JOBS_NO_MEMORY;
     }
 
-    /* Clear the request JSON. */
-    ( void ) memset( pJobsRequest, 0x00, requestLength );
-
-    /* Construct the request JSON, which consists of just a clientToken key. */
-    APPEND_STRING( pJobsRequest, copyOffset, "{\"", 2 );
-    copyOffset = _appendClientToken( pJobsRequest, copyOffset, pRequestInfo, pOperation );
-    APPEND_STRING( pJobsRequest, copyOffset, "\"}", 2 );
-
-    /* Set the output parameters. */
-    pOperation->pJobsRequest = pJobsRequest;
-    pOperation->jobsRequestLength = requestLength;
-
-    /* Ensure offsets are valid. */
-    AwsIotJobs_Assert( copyOffset == requestLength );
-    AwsIotJobs_Assert( pOperation->pClientToken > pOperation->pJobsRequest );
-    AwsIotJobs_Assert( pOperation->pClientToken <
-                       pOperation->pJobsRequest + pOperation->jobsRequestLength );
-
-    IotLogDebug( "Jobs GET PENDING request: %.*s",
-                 pOperation->jobsRequestLength,
-                 pOperation->pJobsRequest );
+    if( status == AWS_IOT_JOBS_SUCCESS )
```

Comment:
This `if`, along with several others in this PR, could just be an `else` since `status` is only changed in the previous `if` block.

Reply:
That is a better idea. I will do.